pkgs/development/python-modules/pyrevolve/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, contexttimer
+, versioneer
+, cython
+, numpy
+, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "pyrevolve";
+  version = "2.2";
+
+  src = fetchFromGitHub {
+    owner = "devitocodes";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-5a4zvyf2vfz8aI6vFMI2vxekYrcUi/YuPFvZnUOx+Zs=";
+  };
+
+  nativeBuildInputs = [ versioneer cython ];
+  propagatedBuildInputs = [ contexttimer ];
+
+  checkInputs = [ pytest numpy ];
+  # Using approach bellow bcs the tests fail with the pythonCheckHook
```

Comment:
Can you post a log?

Reply:
I can, one moment

Reply:
```
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing setuptools-check-hook
Using setuptoolsCheckPhase
Sourcing pytest-check-hook
Using pytestCheckPhase
Removing setuptoolsCheckPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/x6qqhyvaa7ad7lpq2n1qqgmz0gn05nyc-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/versioneer.py
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
Executing setuptoolsBuildPhase
running bdist_wheel
Compiling pyrevolve/crevolve.pyx because it changed.
[1/1] Cythonizing pyrevolve/crevolve.pyx
running build
running build_py
creating build
creating build/lib.linux-x86_64-3.9
creating build/lib.linux-x86_64-3.9/pyrevolve
copying pyrevolve/_version.py -> build/lib.linux-x86_64-3.9/pyrevolve
copying pyrevolve/tools.py -> build/lib.linux-x86_64-3.9/pyrevolve
copying pyrevolve/storage.py -> build/lib.linux-x86_64-3.9/pyrevolve
copying pyrevolve/schedulers.py -> build/lib.linux-x86_64-3.9/pyrevolve
copying pyrevolve/pyrevolve.py -> build/lib.linux-x86_64-3.9/pyrevolve
copying pyrevolve/profiling.py -> build/lib.linux-x86_64-3.9/pyrevolve
copying pyrevolve/logger.py -> build/lib.linux-x86_64-3.9/pyrevolve
copying pyrevolve/custom_pickle.py -> build/lib.linux-x86_64-3.9/pyrevolve
copying pyrevolve/compression.py -> build/lib.linux-x86_64-3.9/pyrevolve
copying pyrevolve/__init__.py -> build/lib.linux-x86_64-3.9/pyrevolve
UPDATING build/lib.linux-x86_64-3.9/pyrevolve/_version.py
set build/lib.linux-x86_64-3.9/pyrevolve/_version.py to '2.2'
running build_ext
building 'pyrevolve.crevolve' extension
creating build/temp.linux-x86_64-3.9
creating build/temp.linux-x86_64-3.9/pyrevolve
creating build/temp.linux-x86_64-3.9/src
g++ -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I./pyrevolve -I. -Ipyrevolve -I/nix/store/hym1n0ygqp9wcm7pxn4sfrql3fg7xa09-python3-3.9.12/include/python3.9 -c pyrevolve/crevolve.cpp -o build/temp.linux-x86_64-3.9/pyrevolve/crevolve.o
In file included from [01m[Kpyrevolve/../include/revolve_c.h:6[m[K,
                 from [01m[Kpyrevolve/crevolve.cpp:722[m[K:
[01m[Kpyrevolve/../include/revolve.h:[m[K In destructor ‘[01m[KRevolve::~Revolve()[m[K’:
[01m[Kpyrevolve/../include/revolve.h:441:22:[m[K [01;35m[Kwarning: [m[Kdeleting object of polymorphic class type ‘[01m[KSchedule[m[K’ which has non-virtual destructor might cause undefined behavior [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdelete-non-virtual-dtor-Wdelete-non-virtual-dtor]8;;[m[K]
  441 |         ~Revolve() { [01;35m[Kdelete f[m[K, delete checkpoint; }
      |                      [01;35m[K^~~~~~~~[m[K
g++ -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I./pyrevolve -I. -Ipyrevolve -I/nix/store/hym1n0ygqp9wcm7pxn4sfrql3fg7xa09-python3-3.9.12/include/python3.9 -c src/revolve.cpp -o build/temp.linux-x86_64-3.9/src/revolve.o
In file included from [01m[Ksrc/revolve.cpp:23[m[K:
[01m[Ksrc/../include/revolve.h:[m[K In destructor ‘[01m[KRevolve::~Revolve()[m[K’:
[01m[Ksrc/../include/revolve.h:441:22:[m[K [01;35m[Kwarning: [m[Kdeleting object of polymorphic class type ‘[01m[KSchedule[m[K’ which has non-virtual destructor might cause undefined behavior [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdelete-non-virtual-dtor-Wdelete-non-virtual-dtor]8;;[m[K]
  441 |         ~Revolve() { [01;35m[Kdelete f[m[K, delete checkpoint; }
      |                      [01;35m[K^~~~~~~~[m[K
[01m[Ksrc/revolve.cpp:[m[K In member function ‘[01m[KACTION::action Revolve::revolve(int*, int*, int*, int, int*)[m[K’:
[01m[Ksrc/revolve.cpp:1247:5:[m[K [01;35m[Kwarning: [m[Kdeleting object of polymorphic class type ‘[01m[KSchedule[m[K’ which has non-virtual destructor might cause undefined behavior [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdelete-non-virtual-dtor-Wdelete-non-virtual-dtor]8;;[m[K]
 1247 |     [01;35m[Kdelete f[m[K;
      |     [01;35m[K^~~~~~~~[m[K
[01m[Ksrc/revolve.cpp:1254:5:[m[K [01;35m[Kwarning: [m[Kdeleting object of polymorphic class type ‘[01m[KSchedule[m[K’ which has non-virtual destructor might cause undefined behavior [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdelete-non-virtual-dtor-Wdelete-non-virtual-dtor]8;;[m[K]
 1254 |     [01;35m[Kdelete f[m[K;
      |     [01;35m[K^~~~~~~~[m[K
[01m[Ksrc/revolve.cpp:[m[K In member function ‘[01m[KACTION::action Revolve::revolve()[m[K’:
[01m[Ksrc/revolve.cpp:1273:5:[m[K [01;35m[Kwarning: [m[Kdeleting object of polymorphic class type ‘[01m[KSchedule[m[K’ which has non-virtual destructor might cause undefined behavior [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdelete-non-virtual-dtor-Wdelete-non-virtual-dtor]8;;[m[K]
 1273 |     [01;35m[Kdelete f[m[K;
      |     [01;35m[K^~~~~~~~[m[K
[01m[Ksrc/revolve.cpp:1280:5:[m[K [01;35m[Kwarning: [m[Kdeleting object of polymorphic class type ‘[01m[KSchedule[m[K’ which has non-virtual destructor might cause undefined behavior [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdelete-non-virtual-dtor-Wdelete-non-virtual-dtor]8;;[m[K]
 1280 |     [01;35m[Kdelete f[m[K;
      |     [01;35m[K^~~~~~~~[m[K
[01m[Ksrc/revolve.cpp:[m[K In member function ‘[01m[Kvoid Revolve::turn(int)[m[K’:
[01m[Ksrc/revolve.cpp:1304:5:[m[K [01;35m[Kwarning: [m[Kdeleting object of polymorphic class type ‘[01m[KSchedule[m[K’ which has non-virtual destructor might cause undefined behavior [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdelete-non-virtual-dtor-Wdelete-non-virtual-dtor]8;;[m[K]
 1304 |     [01;35m[Kdelete f[m[K;
      |     [01;35m[K^~~~~~~~[m[K
[01m[Ksrc/revolve.cpp:1306:5:[m[K [01;35m[Kwarning: [m[Kdeleting object of polymorphic class type ‘[01m[KOnline[m[K’ which has non-virtual destructor might cause undefined behavior [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdelete-non-virtual-dtor-Wdelete-non-virtual-dtor]8;;[m[K]
 1306 |     [01;35m[Kdelete g[m[K;
      |     [01;35m[K^~~~~~~~[m[K
[01m[Ksrc/revolve.cpp:[m[K In member function ‘[01m[Kint Revolve::get_r(int, int)[m[K’:
[01m[Ksrc/revolve.cpp:1400:20:[m[K [01;35m[Kwarning: [m[Kunused variable ‘[01m[Knum[m[K’ [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable]8;;[m[K]
 1400 |   int reps, range, [01;35m[Knum[m[K;
      |                    [01;35m[K^~~[m[K
[01m[Ksrc/revolve.cpp:[m[K In member function ‘[01m[Kint Revolve::get_r()[m[K’:
[01m[Ksrc/revolve.cpp:1431:20:[m[K [01;35m[Kwarning: [m[Kunused variable ‘[01m[Knum[m[K’ [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable-Wunused-variable]8;;[m[K]
 1431 |   int reps, range, [01;35m[Knum[m[K;
      |                    [01;35m[K^~~[m[K
g++ -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I./pyrevolve -I. -Ipyrevolve -I/nix/store/hym1n0ygqp9wcm7pxn4sfrql3fg7xa09-python3-3.9.12/include/python3.9 -c src/revolve_c.cpp -o build/temp.linux-x86_64-3.9/src/revolve_c.o
In file included from [01m[Ksrc/../include/revolve_c.h:6[m[K,
                 from [01m[Ksrc/revolve_c.cpp:1[m[K:
[01m[Ksrc/../include/revolve.h:[m[K In destructor ‘[01m[KRevolve::~Revolve()[m[K’:
[01m[Ksrc/../include/revolve.h:441:22:[m[K [01;35m[Kwarning: [m[Kdeleting object of polymorphic class type ‘[01m[KSchedule[m[K’ which has non-virtual destructor might cause undefined behavior [[01;35m[K]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdelete-non-virtual-dtor-Wdelete-non-virtual-dtor]8;;[m[K]
  441 |         ~Revolve() { [01;35m[Kdelete f[m[K, delete checkpoint; }
      |                      [01;35m[K^~~~~~~~[m[K
g++ -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -shared build/temp.linux-x86_64-3.9/pyrevolve/crevolve.o build/temp.linux-x86_64-3.9/src/revolve.o build/temp.linux-x86_64-3.9/src/revolve_c.o -L/nix/store/hym1n0ygqp9wcm7pxn4sfrql3fg7xa09-python3-3.9.12/lib -o build/lib.linux-x86_64-3.9/pyrevolve/crevolve.cpython-39-x86_64-linux-gnu.so
/nix/store/xi8ha1k3mjpmnmmi5fp7fmq41f2wwghm-python3.9-setuptools-61.2.0/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/pyrevolve
copying build/lib.linux-x86_64-3.9/pyrevolve/crevolve.cpython-39-x86_64-linux-gnu.so -> build/bdist.linux-x86_64/wheel/pyrevolve
copying build/lib.linux-x86_64-3.9/pyrevolve/_version.py -> build/bdist.linux-x86_64/wheel/pyrevolve
copying build/lib.linux-x86_64-3.9/pyrevolve/__init__.py -> build/bdist.linux-x86_64/wheel/pyrevolve
copying build/lib.linux-x86_64-3.9/pyrevolve/compression.py -> build/bdist.linux-x86_64/wheel/pyrevolve
copying build/lib.linux-x86_64-3.9/pyrevolve/custom_pickle.py -> build/bdist.linux-x86_64/wheel/pyrevolve
copying build/lib.linux-x86_64-3.9/pyrevolve/logger.py -> build/bdist.linux-x86_64/wheel/pyrevolve
copying build/lib.linux-x86_64-3.9/pyrevolve/profiling.py -> build/bdist.linux-x86_64/wheel/pyrevolve
copying build/lib.linux-x86_64-3.9/pyrevolve/pyrevolve.py -> build/bdist.linux-x86_64/wheel/pyrevolve
copying build/lib.linux-x86_64-3.9/pyrevolve/schedulers.py -> build/bdist.linux-x86_64/wheel/pyrevolve
copying build/lib.linux-x86_64-3.9/pyrevolve/storage.py -> build/bdist.linux-x86_64/wheel/pyrevolve
copying build/lib.linux-x86_64-3.9/pyrevolve/tools.py -> build/bdist.linux-x86_64/wheel/pyrevolve
running install_egg_info
running egg_info
creating pyrevolve.egg-info
writing pyrevolve.egg-info/PKG-INFO
writing dependency_links to pyrevolve.egg-info/dependency_links.txt
writing requirements to pyrevolve.egg-info/requires.txt
writing top-level names to pyrevolve.egg-info/top_level.txt
writing manifest file 'pyrevolve.egg-info/SOURCES.txt'
reading manifest file 'pyrevolve.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
adding license file 'LICENSE'
writing manifest file 'pyrevolve.egg-info/SOURCES.txt'
Copying pyrevolve.egg-info to build/bdist.linux-x86_64/wheel/pyrevolve-2.2-py3.9.egg-info
running install_scripts
adding license file "LICENSE" (matched pattern "LICEN[CS]E*")
creating build/bdist.linux-x86_64/wheel/pyrevolve-2.2.dist-info/WHEEL
creating 'dist/pyrevolve-2.2-cp39-cp39-linux_x86_64.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'pyrevolve/__init__.py'
adding 'pyrevolve/_version.py'
adding 'pyrevolve/compression.py'
adding 'pyrevolve/crevolve.cpython-39-x86_64-linux-gnu.so'
adding 'pyrevolve/custom_pickle.py'
adding 'pyrevolve/logger.py'
adding 'pyrevolve/profiling.py'
adding 'pyrevolve/pyrevolve.py'
adding 'pyrevolve/schedulers.py'
adding 'pyrevolve/storage.py'
adding 'pyrevolve/tools.py'
adding 'pyrevolve-2.2.dist-info/LICENSE'
adding 'pyrevolve-2.2.dist-info/METADATA'
adding 'pyrevolve-2.2.dist-info/WHEEL'
adding 'pyrevolve-2.2.dist-info/top_level.txt'
adding 'pyrevolve-2.2.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Finished executing setuptoolsBuildPhase
@nix { "action": "setPhase", "phase": "installPhase" }
installing
Executing pipInstallPhase
/build/source/dist /build/source
Processing ./pyrevolve-2.2-cp39-cp39-linux_x86_64.whl
Requirement already satisfied: contexttimer in /nix/store/0r5w3wkrn733cvxww8qgbmxv0lg5i8sa-python3.9-contexttimer-0.3.3/lib/python3.9/site-packages (from pyrevolve==2.2) (0.3.3)
Installing collected packages: pyrevolve
Successfully installed pyrevolve-2.2
/build/source
Finished executing pipInstallPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/q8gbkvp3qhkd6ws2vlqcgmq2c5wy1m4k-python3.9-pyrevolve-2.2
shrinking /nix/store/q8gbkvp3qhkd6ws2vlqcgmq2c5wy1m4k-python3.9-pyrevolve-2.2/lib/python3.9/site-packages/pyrevolve/crevolve.cpython-39-x86_64-linux-gnu.so
strip is /nix/store/58pwclg9yr437h0pfgrnbd0jis8fqasd-gcc-wrapper-11.2.0/bin/strip
stripping (with command strip and flags -S) in /nix/store/q8gbkvp3qhkd6ws2vlqcgmq2c5wy1m4k-python3.9-pyrevolve-2.2/lib 
patching script interpreter paths in /nix/store/q8gbkvp3qhkd6ws2vlqcgmq2c5wy1m4k-python3.9-pyrevolve-2.2
checking for references to /build/ in /nix/store/q8gbkvp3qhkd6ws2vlqcgmq2c5wy1m4k-python3.9-pyrevolve-2.2...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
@nix { "action": "setPhase", "phase": "installCheckPhase" }
running install tests
no Makefile or custom installCheckPhase, doing nothing
@nix { "action": "setPhase", "phase": "pythonCatchConflictsPhase" }
pythonCatchConflictsPhase
@nix { "action": "setPhase", "phase": "pythonRemoveBinBytecodePhase" }
pythonRemoveBinBytecodePhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: pyrevolve
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
pytestCheckPhase
Executing pytestCheckPhase
[1m============================= test session starts ==============================[0m
platform linux -- Python 3.9.12, pytest-7.1.1, pluggy-1.0.0
rootdir: /build/source
[1mcollecting ... [0m[1m
collected 0 items / 3 errors                                                   [0m

==================================== ERRORS ====================================
[31m[1m__________________ ERROR collecting tests/test_compression.py __________________[0m
[31mImportError while importing test module '/build/source/tests/test_compression.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/hym1n0ygqp9wcm7pxn4sfrql3fg7xa09-python3-3.9.12/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_compression.py:4: in <module>
    from pyrevolve.compression import (compressors, decompressors,
pyrevolve/__init__.py:3: in <module>
    from pyrevolve.pyrevolve import * # noqa
pyrevolve/pyrevolve.py:6: in <module>
    from . import crevolve as cr
E   ImportError: cannot import name 'crevolve' from partially initialized module 'pyrevolve' (most likely due to a circular import) (/build/source/pyrevolve/__init__.py)[0m
[31m[1m___________________ ERROR collecting tests/test_pyrevolve.py ___________________[0m
[31mImportError while importing test module '/build/source/tests/test_pyrevolve.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/hym1n0ygqp9wcm7pxn4sfrql3fg7xa09-python3-3.9.12/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_pyrevolve.py:1: in <module>
    from utils import SimpleOperator, SimpleCheckpoint
tests/utils.py:1: in <module>
    from pyrevolve import Operator, Checkpoint
pyrevolve/__init__.py:3: in <module>
    from pyrevolve.pyrevolve import * # noqa
pyrevolve/pyrevolve.py:6: in <module>
    from . import crevolve as cr
E   ImportError: cannot import name 'crevolve' from partially initialized module 'pyrevolve' (most likely due to a circular import) (/build/source/pyrevolve/__init__.py)[0m
[31m[1m____________________ ERROR collecting tests/test_storage.py ____________________[0m
[31mImportError while importing test module '/build/source/tests/test_storage.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/hym1n0ygqp9wcm7pxn4sfrql3fg7xa09-python3-3.9.12/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_storage.py:1: in <module>
    from pyrevolve.compression import init_compression, compressors_available
pyrevolve/__init__.py:3: in <module>
    from pyrevolve.pyrevolve import * # noqa
pyrevolve/pyrevolve.py:6: in <module>
    from . import crevolve as cr
E   ImportError: cannot import name 'crevolve' from partially initialized module 'pyrevolve' (most likely due to a circular import) (/build/source/pyrevolve/__init__.py)[0m
=========================== short test summary info ============================
ERROR tests/test_compression.py
ERROR tests/test_pyrevolve.py
ERROR tests/test_storage.py
!!!!!!!!!!!!!!!!!!! Interrupted: 3 errors during collection !!!!!!!!!!!!!!!!!!!!
[31m============================== [31m[1m3 errors[0m[31m in 0.26s[0m[31m ===============================[0m
```

Reply:
Can you add ``ImportError: cannot import name 'crevolve' from partially initialized module 'pyrevolve' (most likely due to a circular import)`` to the comment?

Reply:
done!